great_expectations/exceptions/exceptions.py

```diff
@@ -65,6 +69,14 @@ def __init__(self, name: str) -> None:
         )
 
 
+class ExpectationSuiteNotFreshError(ResourceNotAddedError):
```

Comment:
Each of suite, batch def, validation def, and checkpoint should have two error types:
- not added
- not fresh

They should all be derived from the same base class (probably `ResourceNotFreshError`?)